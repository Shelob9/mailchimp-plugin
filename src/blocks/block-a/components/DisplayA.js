import React from 'react';

export const DisplayA = ({message,useBold = false}) => {
	if( useBold ){
		return <span><strong>{message}</strong></span>
	}

	return <span>{message}</span>
};

