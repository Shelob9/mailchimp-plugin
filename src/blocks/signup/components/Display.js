import React from 'react';
import {mailChimpTestForm} from "../../../components/mailChimpTestForm.fixture";
import MailChimpForm from "../../../components/MailChimpForm";

export const Display = ({listId,form,Fallback}) => {
	if( listId ){
		return <MailChimpForm
			form={form}
			onSubmit={(values) => alert(JSON.stringify(values))}
		/>
	}

	return <Fallback/>
};

