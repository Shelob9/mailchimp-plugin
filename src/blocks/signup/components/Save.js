import {createElement} from '@wordpress/element';
import {blockClassNameIdentifiers} from "../../blockClassNameIdentifiers";
export const Save = ({attributes, className}) =>  (
    <div className={className}>
        <span className={blockClassNameIdentifiers.signup} data-list={attributes.listId}></span>
    </div>
);