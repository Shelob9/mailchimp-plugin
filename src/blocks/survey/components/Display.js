import {createElement} from '@wordpress/element';
import CalderaMailChimpSurveyForm from "../../../components/CalderaMailChimpSurveyForm";



export const Display = ({listId, apiRoot,token,Fallback}) => {
	if( listId && '---' !== listId ){
		return <CalderaMailChimpSurveyForm
			apiRoot={apiRoot}
			token={token}
			listId={listId}
			onSubmit={(values) => alert(JSON.stringify(values))}
		/>

	}
	return <Fallback/>
};


