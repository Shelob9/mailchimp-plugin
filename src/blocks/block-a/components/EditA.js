import React from 'react';
import {TextControl,RadioControl} from '@wordpress/components';
export const EditA = ({message,onChangeMessage,useBold,onChangeBold}) =>
	 (
		<div>
			<TextControl
				className={'edit-a-message'}
				label={'The Message'}
				value={message}
				onChange={onChangeMessage}
			/>
			<RadioControl
				className={'edit-a-use-bold'}
				options={ [
					{ label: 'Bold', value: true },
					{ label: 'No Special Style', value: false },
				] }
				label={'Use Bold'}
				value={useBold}
				onChange={(useBold) => {
					onChangeBold((useBold == 'true'))
				}}
			/>
		</div>
	);


