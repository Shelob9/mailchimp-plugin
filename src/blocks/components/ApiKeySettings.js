import {Component, Fragment} from "@wordpress/element";
import {Field} from '@calderajs/components';
export class ApiKeySettings extends Component {


    constructor(props) {
        super(props);
        this.state = {
            apiKey: ''
        };
        this.setApiKey = this.setApiKey.bind(this);
        this.onSaveApiKey = this.onSaveApiKey.bind(this);
    }


    setApiKey(apiKey) {
        this.setState({apiKey})
    }

    onSaveApiKey() {
        this.props.adminApiClient.saveApiKey(this.state.apiKey);
        this.setState({apiKey: ''});
    }

    render() {
        const {instanceId} = this.props;
        const {apiKey} = this.state;

        return (

            <Fragment>
                <Field
                    field={{
                        fieldType: 'text',
                        value: apiKey,
                        label: 'New API Key',
                        fieldId: 'caldera-mc-api-key',
                        required: true
                    }}
                    onChange={this.setApiKey}
                    value={apiKey}
                    instanceId={`caldera-mc-select-${instanceId}`}
                />
                <button
                    onClick={this.onSaveApiKey}
                    id={`${instanceId}-mc-add-api-key`}

                    title={'Save API Key'}
                >
                    Save API Key
                </button>

            </Fragment>
        );
    }


}
