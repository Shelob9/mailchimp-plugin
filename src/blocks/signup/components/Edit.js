import React, {Fragment} from 'react';
import {SelectList} from "../../../components/Admin/SelectList/SelectList";
import {Field} from '@calderajs/components';

class ApiKeySettings extends React.Component {


	constructor(props) {
		super(props);
		this.state = {
			apiKey: ''
		};
		this.setApiKey = this.setApiKey.bind(this);
		this.onSaveApiKey = this.onSaveApiKey.bind(this);
	}


	setApiKey(apiKey) {
		this.setState({apiKey})
	}
	onSaveApiKey(){
		this.props.adminApiClient.saveApiKey(this.state.apiKey);
		this.setState({apiKey:''});
	}

	render() {
		const {instanceId} = this.props;
		const {apiKey} = this.state;
		return (

			<Fragment>
				<Field
					field={{
						fieldType: 'text',
						value: apiKey,
						label: 'New API Key',
						fieldId: 'caldera-mc-api-key',
						required: true
					}}
					onChange={this.setApiKey}
					value={apiKey}
					instanceId={`caldera-mc-select-${instanceId}`}
				/>
				<button
					onClick={this.onSaveApiKey}
					id={`${instanceId}-mc-add-api-key`}

					title={'Save API Key'}
				>
					Save API Key
				</button>

			</Fragment>
		);
	}


}


export const Edit = (
	{
		accountId,
		chooseAccountField,
		onChangeAccountId,
		adminApiClient,
		setListId, listId, listFieldConfig, instanceId
	}
) => {
	const initialValues = {};

	function getListIdFromValues(values) {
		return '';
	}

	if (!chooseAccountField.hasOwnProperty('options')) {
		chooseAccountField.options = [];
	}
	let hasOptions =false;
	if (!chooseAccountField.options.length) {
		chooseAccountField.options.push({
			value: null,
			label: '--'
		})
	}else{
		hasOptions = true;
	}
	return (
		<Fragment>
			<ApiKeySettings
				adminApiClient={adminApiClient}
				instanceId={instanceId}
			/>
			{hasOptions &&
				<Field
					field={{
						...chooseAccountField,
						value: accountId,
						fieldId: `${chooseAccountField.fieldId}-${instanceId}`
					}}
					onChange={(newValue) => {
						onChangeAccountId(newValue);
					}}
				/>
			}

			{accountId &&
				<SelectList
					listFieldConfig={listFieldConfig[0]}
					listId={listId}
					instanceId={instanceId}
					setListId={setListId}
				/>
			}

		</Fragment>
	);
}



