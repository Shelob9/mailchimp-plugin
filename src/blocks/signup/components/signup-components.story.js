import React from 'react';
import {storiesOf} from '@storybook/react';
import {Button} from '@storybook/react/demo';
import {Display} from "./Display";
import {Edit} from "./Edit";
import {mailChimpTestForm} from "../../../components/mailChimpTestForm.fixture";
import {SignupBlockEdit} from "../index";

const listId = 'ff';
const attributes = {
	listId
};
const setAttributes = (newValues) => console.log(newValues);
const instanceId = 'instanceIdIsThis';
storiesOf('The signup block', module)
	.add('Signup Block Edit UI', () => (
		<SignupBlockEdit
			attributes={attributes} setAttributes={setAttributes} instanceId={instanceId}
		/>
	))
	.add('Display component', () => (
		<Display
			listId={listId}
			form={mailChimpTestForm}
			Fallback={() => (<div>Choose</div>)}
		/>
	))
	.add('Edit Component', () => (
		<Edit
			listId={'listId'}
			onChangeListId={(values) => console.log(values)}
			listFields={[]}//@todo mock for this!
			instanceId={'foo'}
		/>
	));
