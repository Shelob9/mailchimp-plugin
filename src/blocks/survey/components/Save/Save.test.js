import {Save} from "./Save";
import renderer from 'react-test-renderer'
import {Display} from "../../../signup/components/Display";
import {mailChimpTestForm} from "../../../../components/mailChimpTestForm.fixture";
test( 'Save matches snapshot', ()=> {
    const attributes= {
        listId:'f411',
        accountId: 3
    };
    const component = renderer.create(
        Save({className:'the-classname',attributes})
    );
    expect(component.toJSON()).toMatchSnapshot();

});