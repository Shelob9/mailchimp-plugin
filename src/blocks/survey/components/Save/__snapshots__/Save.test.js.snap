// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save matches snapshot 1`] = `
<div
  className="the-classname"
>
  <span
    className="calderaMailchimpSurvery"
    data-list="f411"
  />
</div>
`;
